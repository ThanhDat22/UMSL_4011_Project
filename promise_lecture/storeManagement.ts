export interface Inventory {
    name: string;
    quantity: number;
    pricePerUnit: number;
    wholesalePrice: number;
    saleTaxPerUnit?: number;
}

export const getValueOfStore = (item: Inventory[]): number => {
    const totalValue = item.reduce((acc, item) => {
        return acc + (item.quantity * item.pricePerUnit);
    }, 0);
    return totalValue;
}

export interface InventoryWithProfiPotential extends Inventory {
    potentialProfit: number;
}

export const getInventoryWithProfitPotential = (item: Inventory[]): InventoryWithProfiPotential[] => {
    const itemWithProfitPotential: InventoryWithProfiPotential[] = item.map((item: Inventory) => {




        const result: InventoryWithProfiPotential = {
            ...item,
            potentialProfit: item.pricePerUnit - item.wholesalePrice
        };
        return result;
    });
    return itemWithProfitPotential;


}
