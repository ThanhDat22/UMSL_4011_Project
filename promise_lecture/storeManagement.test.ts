import { Inventory, getValueOfStore, getInventoryWithProfitPotential } from "./storeManagement";

describe('getValueOfStore', () => {
    it('Adds up the value of all items in the store', () => {
        const items: Inventory[] = [
            {
                name: 'iPad Pro',
                quantity: 5,
                pricePerUnit: 1200,
                wholesalePrice: 1000
            },
            {
                name: 'Macbook Pro',
                quantity: 7,
                pricePerUnit: 1700,
                wholesalePrice: 1500
            }
        ];
        const actual = getValueOfStore(items);
        const expected = 17900;
        expect(actual).toEqual(expected);
    });

    it('Adds up the value of all items in the store', () => {
        const items: Inventory[] = [
            {
                name: 'iPad Pro',
                quantity: 8,
                pricePerUnit: 1200,
                wholesalePrice: 1000
            },
            {
                name: 'Macbook Pro',
                quantity: 2,
                pricePerUnit: 1700,
                wholesalePrice: 1500
            }
        ];
        const actual = getValueOfStore(items);
        const expected = 13000;
        expect(actual).toEqual(expected);
    });

});

describe('getInventoryWithProfitPotential', () => {
    it('Generates an item array with Profit Potential', () => {
        const items: Inventory[] = [
            {
                name: 'iPad Pro',
                quantity: 8,
                pricePerUnit: 1200,
                wholesalePrice: 1000
            },
            {
                name: 'Macbook Pro',
                quantity: 2,
                pricePerUnit: 1700,
                wholesalePrice: 1500
            }
        ];

        const actual = getInventoryWithProfitPotential(items);
        expect(actual).toMatchSnapshot();
    });
});
