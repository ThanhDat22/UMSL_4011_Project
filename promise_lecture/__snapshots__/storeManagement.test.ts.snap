// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getInventoryWithProfitPotential Generates an item array with Profit Potential 1`] = `
[
  {
    "name": "iPad Pro",
    "potentialProfit": 200,
    "pricePerUnit": 1200,
    "quantity": 8,
    "wholesalePrice": 1000,
  },
  {
    "name": "Macbook Pro",
    "potentialProfit": 200,
    "pricePerUnit": 1700,
    "quantity": 2,
    "wholesalePrice": 1500,
  },
]
`;
